ledger-core/virtual/integration/method_test.go

```diff
@@ -1993,3 +1993,236 @@ func TestVirtual_Method_IntolerableCallChangeState(t *testing.T) {
+				Type:   execution.Done,
+				Result: requestresult.New([]byte("get validated info 2"), gen.UniqueGlobalRef()),
+			},
+		)
+		runnerMock.AddExecutionMock("GetDirtyMethod2").AddStart(
+			func(ctx execution.Context) {
+				assert.Equal(t, contract.CallDirty, ctx.Isolation.State)
+				assert.Equal(t, newState, ctx.ObjectDescriptor.Memory())
+			},
+			&execution.Update{
+				Type:   execution.Done,
+				Result: requestresult.New([]byte("get dirty info 2"), gen.UniqueGlobalRef()),
+			},
+		)
+		runnerMock.AddExecutionClassify("GetValidatedMethod1", intolerableFlags(), nil)
+		runnerMock.AddExecutionClassify("GetValidatedMethod2", intolerableFlags(), nil)
+		runnerMock.AddExecutionClassify("GetDirtyMethod1", contract.MethodIsolation{Interference: contract.CallIntolerable, State: contract.CallDirty}, nil)
+		runnerMock.AddExecutionClassify("GetDirtyMethod2", contract.MethodIsolation{Interference: contract.CallIntolerable, State: contract.CallDirty}, nil)
+	}
+
+	// add typedChecker mock
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+	{
+		typedChecker.VStateReport.Set(func(report *payload.VStateReport) bool {
+			assert.Equal(t, newState, report.ProvidedContent.LatestDirtyState.State)
+			assert.Equal(t, newState, report.ProvidedContent.LatestValidatedState.State)
+			return false
+		})
+		typedChecker.VCallResult.Set(func(result *payload.VCallResult) bool {
+			switch string(result.ReturnArguments) {
+			case "get validated info 1":
+			case "get validated info 2":
+			case "get dirty info 1":
+			case "get dirty info 2":
+			case "done":
+			default:
+				t.Fatalf("unexpected result")
+			}
+			return false
+		})
+	}
+
+	// initial object state
+	{
+		// need for correct handle state report (should from prev pulse)
+		server.IncrementPulse(ctx)
+
+		report := &payload.VStateReport{
+			Status: payload.Ready,
+			Object: objectGlobal,
+			AsOf:   prevPulse,
+			ProvidedContent: &payload.VStateReport_ProvidedContentBody{
+				LatestDirtyState: &payload.ObjectState{
+					Reference: reference.Local{},
+					Class:     class,
+					State:     initialState,
+				},
+				LatestValidatedState: &payload.ObjectState{
+					Reference: reference.Local{},
+					Class:     class,
+					State:     initialState,
+				},
+			},
+		}
+		waitReport := server.Journal.WaitStopOf(&handlers.SMVStateReport{}, 1)
+		server.SendPayload(ctx, report)
+		commontestutils.WaitSignalsTimed(t, 10*time.Second, waitReport)
+	}
+
+	// get object state
+	{
+		executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
+		pl := &payload.VCallRequest{
+			CallType:            payload.CTMethod,
+			CallFlags:           payload.BuildCallFlags(contract.CallIntolerable, contract.CallValidated),
+			Caller:              server.GlobalCaller(),
+			Callee:              objectGlobal,
+			CallSiteDeclaration: class,
+			CallSiteMethod:      "GetValidatedMethod1",
+			CallOutgoing:        gen.UniqueGlobalRefWithPulse(server.GetPulse().PulseNumber),
```

Comment:
can use `outgoing     = server.BuildRandomOutgoingWithPulse()`

Reply:
in all messages with outgoing

Reply:
I don't want to use the same outgoing ref for all calls, because in real life these outgoings won't be the same if I'm right.

Reply:
no, I meant replacing `gen.UniqueGlobalRefWithPulse(server.GetPulse().PulseNumber)` -> `server.BuildRandomOutgoingWithPulse()`

Reply:
done